Angular/src/app/app.component.ts

```diff
@@ -0,0 +1,56 @@
+import { Component } from '@angular/core';
+import { rrulestr } from 'rrule';
+import {CloseButtonOptions} from './interfaces';
+import {appointments, Appointment} from './data/appointments'
+
+@Component({
+  selector: 'demo-app',
+  templateUrl: 'app.component.html',
+  styleUrls: ['app.component.css'],
+})
+
+export class AppComponent {
+  dataSource: Appointment[] = appointments;
+  currentDate: Date = new Date(2022, 9, 1);
+  popupVisible: boolean = false;
+  closeButtonOptions: CloseButtonOptions  = {
+    text: 'Close',
+    onClick: () => {
+      this.popupVisible = false;
+    }
+  };
+
+  checkDateInArray(dateString: string): number {
+    const weekDayNumbers = [6, 0, 1, 2, 3, 4, 5];
+    let date = new Date(dateString);
+    let day = date.getDay();
+
+    return weekDayNumbers[day];
+  }
```

Comment:
1. Let's rename this method to ```getDateDay```
2. Make it ```private``` and move it after public methods
3. Change ```let``` to the ```const``` because we don't change these variables
4. Change ```dateString``` type to ```Date | string``` and rename this argument to ```date``` because 
```AppointmentAddingEvent.appointmentData.(startDate / endDate)``` has this type
5. Move ```weekDayNumbers``` to the const before class initialization

```typescript
// app.component.ts
/* imports here */

const WEEK_DAY_NUMBERS = [6, 0, 1, 2, 3, 4, 5];

@Component({...})
export class AppComponent {...}
```

```suggestion
private getDateDay(appointmentDate: Date | string): number {
    return WEEK_DAY_NUMBERS[new Date(appointmentDate).getDay()];
  }
```